e2e-tests/smart-update/run

```diff
@@ -0,0 +1,195 @@
+
+    wait_cluster_consistency ${cluster} ${pxc_size} $(kubectl_bin get pxc ${cluster} -o jsonpath='{.spec.proxysql.size}')
+    for i in $(seq 0 $((${pxc_size} - 1))); do
+        compare_mysql_cmd "select-1" "SELECT * from myApp.myApp;" "-h $cluster-pxc-$i.$cluster-pxc -uroot -proot_password"
+    done
+
+    kubectl_bin delete -f $test_dir/conf/$cluster.yml
+    kubectl_bin delete pvc --all
+
+    desc 'Updating HAProxy PXC cluster'
+    spinup_pxc "$cluster" "$test_dir/conf/$cluster-haproxy.yml" 3 10 haproxy
+    pxc_size=$(kubectl_bin get pxc "${cluster}" -o jsonpath='{.spec.pxc.size}')
+
+    initial_primary=$(run_mysql 'SELECT @@hostname hostname;' "-h $cluster-haproxy -uroot -proot_password")
+    kubectl_bin patch pxc/$cluster --type=merge -p '{"spec":{"pxc":{"image":"'"${IMAGE_TO_UPDATE}"'"}}}'
+    sleep 7 # wait for two reconcile loops ;)  3 sec x 2 times + 1 sec = 7 seconds
+    
+    check_last_pod_to_update ${cluster} ${initial_primary} ${pxc_size} ${IMAGE_TO_UPDATE}
+    wait_cluster_consistency ${cluster} ${pxc_size} $(kubectl_bin get pxc ${cluster} -o jsonpath='{.spec.haproxy.size}') "haproxy"
+    for i in $(seq 0 $((${pxc_size} - 1))); do
+        compare_mysql_cmd "select-1" "SELECT * from myApp.myApp;" "-h $cluster-pxc-$i.$cluster-pxc -uroot -proot_password"
+    done
+
+    kubectl_bin delete -f "$test_dir/conf/$cluster-haproxy.yml"
+    kubectl_bin delete pvc --all
+
+    desc 'Updating PXC cluster with version service available but disabled'
+    spinup_pxc "$cluster" "$test_dir/conf/$cluster-version-service-reachable.yml"
+    pxc_size=$(kubectl_bin get pxc "${cluster}" -o jsonpath='{.spec.pxc.size}')
+
+    initial_primary=$(get_proxy_primary "-h127.0.0.1 -P6032 -uproxyadmin -padmin_password" "$cluster-proxysql-0")
+    kubectl_bin patch pxc/$cluster --type=merge -p '{"spec":{"pxc":{"image":"'"${IMAGE_TO_UPDATE}"'"}}}'
+    sleep 7 # wait for two reconcile loops ;)  3 sec x 2 times + 1 sec = 7 seconds
+    
+    check_last_pod_to_update ${cluster} ${initial_primary} ${pxc_size} ${IMAGE_TO_UPDATE}
+
+    wait_cluster_consistency ${cluster} ${pxc_size} $(kubectl_bin get pxc ${cluster} -o jsonpath='{.spec.proxysql.size}')
+    for i in $(seq 0 $((${pxc_size} - 1))); do
+        compare_mysql_cmd "select-1" "SELECT * from myApp.myApp;" "-h ${cluster}-pxc-${i}.${cluster}-pxc -uroot -proot_password"
+    done
+
+    kubectl_bin delete -f $test_dir/conf/$cluster-version-service-reachable.yml
+    kubectl_bin delete pvc --all
+
+    desc 'PXC cluster with version service offline'
+    spinup_pxc "$cluster" "$test_dir/conf/$cluster-version-service-unreachable.yml"
+    pxc_size=$(kubectl_bin get pxc "${cluster}" -o jsonpath='{.spec.pxc.size}')
+
+    initial_primary=$(get_proxy_primary "-h127.0.0.1 -P6032 -uproxyadmin -padmin_password" "$cluster-proxysql-0")
+    kubectl_bin patch pxc/$cluster --type=merge -p '{"spec":{"pxc":{"image":"'"${IMAGE_TO_UPDATE}"'"}}}'
```

Comment:
can we for "version service offline" test check that image still equal IMAGE_PXC and not IMAGE_PXC_TO_UPDATE

Reply:
done